src/main/java/ratpack/zipkin/internal/RatpackCurrentTraceContext.java

```diff
@@ -0,0 +1,41 @@
+package ratpack.zipkin.internal;
+
+import brave.internal.Nullable;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import org.slf4j.MDC;
+
+public final class RatpackCurrentTraceContext extends CurrentTraceContext {
+
+  private final MDCProxy mdc;
+
+  private final TraceContext.Extractor<MDCProxy> extractor = Propagation.B3_STRING.extractor(MDCProxy::get);
+  private final TraceContext.Injector<MDCProxy> injector = Propagation.B3_STRING.injector(MDCProxy::put);
+
+  public RatpackCurrentTraceContext() {
+    this.mdc = new MDCProxy();
+  }
+
+  @Override
+  @Nullable
+  public TraceContext get() {
+    return extractor.extract(mdc).context();
+  }
+
+  @Override
+  public Scope newScope(TraceContext currentSpan) {
+    final TraceContext previous = get();
+    injector.inject(currentSpan, mdc);
+    return () -> injector.inject(previous, mdc);
+  }
+
+  private static final class MDCProxy {
```

Comment:
Yeah thats exactly how I ended up with this :)  I don't think the MDC approach is the ideal approach with Ratpack.  It might be useful to put the trace context in MDC for other reasons but I think we can do without it for now.